### ISSUE TITLE
Error: docker conflicts with origin-1.1.4

### ISSUE THREAD

**Issue by david-strejc** [openshift/openshift-ansible#1694]
I got this error:

Error: docker conflicts with origin-1.1.4

When I wan't to install fresh open shift origin. 

This si full error:

`failed: [test01.es3.cz] => {"changed": true, "rc": 1, "results": ["Loaded plugins: fastestmirror\nLoading mirror speeds from cached hostfile\n * base: mirror.karneval.cz\n 
extras: tux.rainside.sk\n * updates: mirror.karneval.cz\nResolving Dependencies\n--> Running transaction check\n---> Package origin-node.x86_64 0:1.1.4-0.git.0.9d8230b.el
7.centos will be installed\n--> Processing Dependency: docker-io >= 1.9.1 for package: origin-node-1.1.4-0.git.0.9d8230b.el7.centos.x86_64\n--> Processing Dependency: socat
 for package: origin-node-1.1.4-0.git.0.9d8230b.el7.centos.x86_64\n---> Package tuned-profiles-origin-node.x86_64 0:1.1.4-0.git.0.9d8230b.el7.centos will be installed\n--> 
Running transaction check\n---> Package docker.x86_64 0:1.8.2-10.el7.centos will be updated\n---> Package docker.x86_64 0:1.9.1-25.el7.centos will be an update\n--> Process
ing Dependency: docker-forward-journald = 1.9.1-25.el7.centos for package: docker-1.9.1-25.el7.centos.x86_64\n---> Package socat.x86_64 0:1.7.2.2-5.el7 will be installed\n-
-> Running transaction check\n---> Package docker-forward-journald.x86_64 0:1.9.1-25.el7.centos will be installed\n--> Processing Conflict: docker-1.9.1-25.el7.centos.x86_6
4 conflicts origin < 1.2\n--> Finished Dependency Resolution\n You could try using --skip-broken to work around the problem\n You could try running: rpm -Va --nofiles --nod
igest\n"]}
`

**Comment by sdodson** [comment]
https://github.com/openshift/openshift-ansible/pull/1655 should improve this though there's some work left to do on that PR. It should be sufficient for now.

Ultimately Origin 1.1.4 and Origin 1.1.5 are fine with docker-1.9 however the conflicts that was put in is for origin < 1.2 and the expectation was that Origin would be bumped to 1.2 prior to docker-1.9 shipping. That never happened unfortunately.

Alternatively you can work around this by pre-installing docker-1.8.2 on all your node hosts.

**Comment by sdodson** [comment]
Actually, that won't do anything for you as origin currently requires docker >= 1.9. I'll build origin-1.1.5 without that requirement.

**Comment by sdodson** [comment]
I've rebuilt origin-1.1.4 because of the critical bug in origin-1.1.5. It should no longer require docker > 1.9. You may have to clear your yum cache on all hosts

`yum clean expire-cache`

**Comment by david-strejc** [comment]
This won't help.

Also during installation there is docker 1.8.2 installed via openshift-ansible.

This is what I got when I try to update my machine.

`
--> Running transaction check
---> Package docker.x86_64 0:1.9.1-0.origin.19.el7.centos will be updated
---> Package docker.x86_64 0:1.9.1-25.el7.centos will be an update
---> Package docker-forward-journald.x86_64 0:1.9.1-0.origin.19.el7.centos will be updated
---> Package docker-forward-journald.x86_64 0:1.9.1-25.el7.centos will be an update
---> Package docker-selinux.x86_64 0:1.9.1-0.origin.19.el7.centos will be updated
---> Package docker-selinux.x86_64 0:1.9.1-25.el7.centos will be an update
---> Package etcd.x86_64 0:2.2.2-5.el7 will be updated
---> Package etcd.x86_64 0:2.2.5-1.el7 will be an update
---> Package origin.x86_64 0:1.1.4-0.git.0.9d8230b.el7.centos will be updated
---> Package origin.x86_64 0:1.1.4-1.git.2.c8de71e.el7.centos will be an update
---> Package origin-clients.x86_64 0:1.1.4-0.git.0.9d8230b.el7.centos will be updated
---> Package origin-clients.x86_64 0:1.1.4-1.git.2.c8de71e.el7.centos will be an update
---> Package origin-node.x86_64 0:1.1.4-0.git.0.9d8230b.el7.centos will be updated
---> Package origin-node.x86_64 0:1.1.4-1.git.2.c8de71e.el7.centos will be an update
--> Processing Dependency: docker-io = 1.8.2 for package: origin-node-1.1.4-1.git.2.c8de71e.el7.centos.x86_64
---> Package origin-sdn-ovs.x86_64 0:1.1.4-0.git.0.9d8230b.el7.centos will be updated
---> Package origin-sdn-ovs.x86_64 0:1.1.4-1.git.2.c8de71e.el7.centos will be an update
---> Package tuned-profiles-origin-node.x86_64 0:1.1.4-0.git.0.9d8230b.el7.centos will be updated
---> Package tuned-profiles-origin-node.x86_64 0:1.1.4-1.git.2.c8de71e.el7.centos will be an update
--> Processing Conflict: docker-1.9.1-25.el7.centos.x86_64 conflicts origin < 1.2
--> Finished Dependency Resolution
Error: docker conflicts with origin-1.1.4-1.git.2.c8de71e.el7.centos.x86_64
Error: Package: origin-node-1.1.4-1.git.2.c8de71e.el7.centos.x86_64 (maxamillion-origin-next)
           Requires: docker-io = 1.8.2
           Removing: docker-1.9.1-0.origin.19.el7.centos.x86_64 (@maxamillion-origin-next)
               docker-io = 1.9.1-0.origin.19.el7.centos
           Updated By: docker-1.9.1-25.el7.centos.x86_64 (extras)
               docker-io = 1.9.1-25.el7.centos
           Available: docker-1.8.2-7.el7.centos.x86_64 (extras)
               docker-io = 1.8.2-7.el7.centos
           Available: docker-1.8.2-8.el7.centos.x86_64 (extras)
               docker-io = 1.8.2-8.el7.centos
           Available: docker-1.8.2-10.el7.centos.x86_64 (extras)
               docker-io = 1.8.2-10.el7.centos
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodiges`

What I've reported was connected to clean installation of openshift origin.

**Comment by detiber** [comment]
@david-strejc how are you doing the installation? Also, can you provide the full ansible output as well?

**Comment by zhwenh** [comment]
# **I also have the same problem, my docker 1.10 has been installed on the server, but according to the following error.**
## failed: [10.1.130.16] => {"changed": true, "rc": 1, "results": ["Loaded plugins: fastestmirror\nLoading mirror speeds from cached hostfile\n * base: mirrors.aliyun.com\n * extras: mirrors.aliyun.com\n * updates: mirrors.aliyun.com\nResolving Dependencies\n--> Running transaction check\n---> Package docker.x86_64 0:1.9.1-25.el7.centos will be installed\n--> Processing Dependency: docker-forward-journald = 1.9.1-25.el7.centos for package: docker-1.9.1-25.el7.centos.x86_64\n--> Processing Dependency: docker-selinux >= 1.9.1-25.el7.centos for package: docker-1.9.1-25.el7.centos.x86_64\n--> Running transaction check\n---> Package docker-forward-journald.x86_64 0:1.9.1-25.el7.centos will be installed\n---> Package docker-selinux.x86_64 0:1.9.1-25.el7.centos will be installed\n--> Processing Conflict: docker-engine-1.10.3-1.el7.centos.x86_64 conflicts docker\n--> Processing Conflict: docker-engine-1.10.3-1.el7.centos.x86_64 conflicts docker-io\n--> Processing Conflict: docker-engine-selinux-1.10.3-1.el7.centos.noarch conflicts docker-selinux\n--> Finished Dependency Resolution\n You could try using --skip-broken to work around the problem\n You could try running: rpm -Va --nofiles --nodigest\n"]}
msg: Error: docker-engine conflicts with docker-1.9.1-25.el7.centos.x86_64
Error: docker-engine-selinux conflicts with docker-selinux-1.9.1-25.el7.centos.x86_64

FATAL: all hosts have already failed -- aborting

**Comment by zhwenh** [comment]
Also ask, is there any solutions by docker-container install openshift orgin, with ansible install too much trouble.

**Comment by sdodson** [comment]
@zhwenh Can you disable whichever repo is giving you docker-1.10 and uninstall all versions of docker then re-try?

**Comment by sdodson** [comment]
I've built origin-1.1.6 and rebuilt docker-1.9.1-25 without the `Conflicts: origin < 1.2`.

**Comment by zhwenh** [comment]
where setting openshift origin installed version?

**Comment by david-strejc** [comment]
I confirm that now update and builds passes. With docker 1.9.25 and open shift 1.1.6

**Comment by zhwenh** [comment]
**I  download the last master branch now ,still show error:**
TASK: [docker | Downgrade docker if necessary] ******************************** 
failed: [10.1.130.14] => {"changed": true, "cmd": ["yum", "downgrade", "-y", "docker-1.8.2"], "delta": "0:00:00.359044", "end": "2016-04-06 12:56:55.767268", "rc": 1, "start": "2016-04-06 12:56:55.408224", "warnings": ["Consider using yum module rather than running yum"]}
stderr: Error: Nothing to do
stdout: Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * epel: ftp.jaist.ac.jp
No package docker-1.8.2 available.

FATAL: all hosts have already failed -- aborting

**Comment by sdodson** [comment]
@zhwenh do you not have centos-extras repo enabled? You shouldn't need epel for docker.

**Comment by zhwenh** [comment]
i add this :
[extras]
name=CentOS-$releasever - Extras
baseurl=http://mirror.centos.org/centos/$releasever/extras/$basearch/
gpgcheck=1
gpgkey=http://mirror.centos.org/centos/RPM-GPG-KEY-CentOS-7

**After manual setup docker 1.9.1 ,now show this errors:**
TASK: [docker | Downgrade docker if necessary] ******************************** 
failed: [10.1.130.14] => {"changed": true, "cmd": ["yum", "downgrade", "-y", "docker-1.9.1"], "delta": "0:00:01.206202", "end": "2016-04-07 09:14:42.796258", "rc": 1, "start": "2016-04-07 09:14:41.590056", "warnings": ["Consider using yum module rather than running yum"]}
stderr: Error: Package: docker-1.9.1-25.el7.centos.x86_64 (extras)
           Requires: docker-forward-journald = 1.9.1-25.el7.centos
           Installed: docker-forward-journald-1.9.1-25.1.origin.el7.centos.x86_64 (@maxamillion-origin-next)
               docker-forward-journald = 1.9.1-25.1.origin.el7.centos
           Available: docker-forward-journald-1.9.1-25.el7.centos.x86_64 (extras)
               docker-forward-journald = 1.9.1-25.el7.centos
stdout: Loaded plugins: fastestmirror
Loading mirror speeds from cached hostfile
 * elrepo: repos.lax-noc.com
 * epel: mirror01.idc.hinet.net
 * rpmforge: mirrors.neusoft.edu.cn
Resolving Dependencies
--> Running transaction check
---> Package docker.x86_64 0:1.9.1-25.el7.centos will be a downgrade
--> Processing Dependency: docker-forward-journald = 1.9.1-25.el7.centos for package: docker-1.9.1-25.el7.centos.x86_64
---> Package docker.x86_64 0:1.9.1-25.1.origin.el7.centos will be erased
--> Finished Dependency Resolution
 You could try using --skip-broken to work around the problem
 You could try running: rpm -Va --nofiles --nodigest

FATAL: all hosts have already failed -- aborting

**Comment by zhwenh** [comment]
The following method can solve the problem:
first  in:    roles\docker_storage_setup\tasks\main.yml
second in:   roles\openshift_docker_facts\tasks\main.yml
let   1.8.2   modifyed to   1.9.1  that's ok!

**Comment by michaelgugino** [comment]
This issue appears to have been solved.

Please consider filing a new issue if the problem persists.
